README.md

```diff
@@ -299,3 +299,150 @@ The `examples` folder contains runnable examples that demonstrate the various wa
 * `examples/authMiddlewares` - Shows usage of the `auth` and `hasRole` middlewares.
 * `subscriptions` - Shows basic subscriptions setup, requiring all subscriptions to be authenticated.
 * `subscriptionsAdvanced` - Shows subscriptions that use the `auth` and `hasRole` middlewares directly on subscription resolvers.
+
+## Setting up the Examples
+
+Prerequisites:
+
+* Docker and docker-compose installed
+* Node.js and NPM installed
+
+Start by cloning this repo.
+
+```
+git clone https://github.com/aerogear/keycloak-connect-graphql/
+```
+
+Then start a Keycloak server using `docker-compose`.
+
+```
+cd examples/config && docker-compose up
+```
+
+Now in a separate terminal, seed the keycloak server with a sample configuration.
+
+```
+$ npm run examples:seed
+
+creating role admin
+creating role developer
+creating client role admin for client keycloak-connect-graphql-bearer
+creating client role developer for client keycloak-connect-graphql-bearer
+creating client role admin for client keycloak-connect-graphql-public
+creating client role developer for client keycloak-connect-graphql-public
+creating user developer with password developer
+assigning client and realm roles called "developer" to user developer
+creating user admin with password admin
+assigning client and realm roles called "admin" to user admin
+done
+```
+
+This creates a sample realm called `keycloak-connect-graphql` with some clients, roles and users that we can use in the examples.
+Now we are ready to start and explore the examples.
+
+The Keycloak console is accessible at [localhost:8080](http://localhost:8080) and the admin login is `admin/admin`. You can make any configuration changes you wish and `npm run examples:seed` will always recreate the example realm from scratch.
+
+## Running the Basic Example
+
+The basic example shows:
+
+* The setup of the keycloak express middleware
+* How to add **Role Based Access Control** using the `@hasRole` schema directive.
+
+In `examples/basic.js` the GraphQL schema for the server is defined:
+
+```js
+const typeDefs = gql`
+  type Query {
+    hello: String @hasRole(role: "developer")
+  }
+`
+```
+
+The `@hasRole` directive means only users with the `developer` role are authorized to perform the `hello` query. Start the server to try it out.
+
+```
+$ node examples/basic.js
+🚀 Server ready at http://localhost:4000/graphql
+```
+
+Open the URL and you will see the Keycloak login screen. First login with `developer/developer` as the username/password.
+
+Now you should see the GraphQL Playground.
+
+NOTE: We initially saw the Keycloak login screen because the keycloak middleware is protecting the `/graphql` endpoint using a `public` client configuration. We're using a public client here because we want to access the GraphQL Playground in our browser. In production your GraphQL API would use a `bearer` client configuration.
```

Comment:
```suggestion
NOTE: We initially going to see the Keycloak login screen because the keycloak middleware will require authentication for the `/graphql` endpoint using a `public` client configuration. We're using a public client here, because we want to access the GraphQL Playground in our browser. In production your GraphQL API would use a `bearer` client configuration.
```